napari/_qt/widgets/qt_viewer_buttons.py

```diff
@@ -195,6 +210,209 @@ def open_perspective_popup(self):
         pop.frame.setLayout(layout)
         pop.show_above_mouse()
 
+    def _open_grid_popup(self):
+        """Open grid options pop up widget."""
+
+        # widgets
+        popup = QtPopup(self)
+        grid_stride = QSpinBox(popup)
+        grid_width = QSpinBox(popup)
+        grid_height = QSpinBox(popup)
+        shape_help_symbol = QLabel(self)
+        stride_help_symbol = QLabel(self)
+        blank = QLabel(self)  # helps with placing help symbols.
+
+        shape_help_msg = trans._(
+            'Number of rows and columns in the grid. A value of -1 for either or '
+            + 'both of will be used the row and column numbers will trigger an '
+            + 'auto calculation of the necessary grid shape to appropriately fill '
+            + 'all the layers at the appropriate stride. '
+            + '0 is not a valid entry.'
+        )
+
+        stride_help_msg = trans._(
+            'Number of layers to place in each grid square before moving on to '
+            + 'the next square. The default ordering is to place the most visible '
+            + 'layer in the top left corner of the grid. A negative stride will '
+            + 'cause the order in which the layers are placed in the grid to be '
+            + 'reversed. '
+            + '0 is not a valid entry.'
+        )
+
+        # set up
+        wh_values = list(range(-1, 10))
+        wh_values = wh_values.pop(1)
+        grid_stride.setObjectName("gridStrideBox")
+        grid_stride.setAlignment(Qt.AlignCenter)
+        grid_stride.setMaximum(20)
+        grid_stride.setMinimum(-20)
```

Comment:
```suggestion
        grid_stride.setRange(-20, 20)
```

https://doc.qt.io/qt-5/qspinbox.html#setRange